includes/specials/SpecialListgrouprights.php

```diff
@@ -113,11 +125,13 @@ public function execute( $par ) {
 				$grouplink = '';
 			}
 
+			$groupArr = $this->userPermissions()->getGroupsChangeableByGroup( $group );
+
 			$revoke = isset( $wgRevokePermissions[$group] ) ? $wgRevokePermissions[$group] : array();
-			$addgroups = isset( $wgAddGroups[$group] ) ? $wgAddGroups[$group] : array();
-			$removegroups = isset( $wgRemoveGroups[$group] ) ? $wgRemoveGroups[$group] : array();
-			$addgroupsSelf = isset( $wgGroupsAddToSelf[$group] ) ? $wgGroupsAddToSelf[$group] : array();
-			$removegroupsSelf = isset( $wgGroupsRemoveFromSelf[$group] ) ? $wgGroupsRemoveFromSelf[$group] : array();
+			$addgroups = isset( $groupArr['add'] ) ? $groupArr['add'] : array();
```

Comment:
how about enforcing `getGroupsChangeableByGroup` to return all those keys and then we could skip checking if they exist here?